ui/src/utils/lineWriter.ts

```diff
@@ -0,0 +1,237 @@
+import AJAX from 'src/utils/ajax'
+
+export interface Tags {
+  [key: string]: string
+}
+
+export interface Fields {
+  [key: string]: number | string
+}
+
+export enum Precision {
+  ns = 'ns',
+  u = 'u',
+  ms = 'ms',
+  s = 's',
+  m = 'm',
+  h = 'h',
+}
+
+export interface BatchOptions {
+  maxIntervalInSeconds: number
+  maxBatchedLines: number
+}
+
+const defaultBatchOptions: BatchOptions = {
+  maxIntervalInSeconds: 10,
+  maxBatchedLines: 100,
+}
+
+const nowInSeconds = function nowInSeconds() {
+  return Math.floor(Date.now() / 1000)
+}
+
+export class LineWriter {
+  protected url: string
+  protected orgId: number | string
+  protected bucketName: string
+  protected authToken: string
+  protected batchOptions: BatchOptions
+
+  protected timerID: number | null
+  protected batchedLines: number
+  protected lines: string[]
+
+  constructor(
+    url: string,
+    orgId: number | string,
+    bucketName: string,
+    authToken: string,
+    batchOptions: BatchOptions = defaultBatchOptions
+  ) {
+    this.url = url
+    this.orgId = orgId
+    this.bucketName = bucketName
+    this.authToken = authToken
+    this.batchOptions = batchOptions
+
+    this.timerID = null
+    this.batchedLines = 0
+    this.lines = []
+  }
+
+  public createLineFromModel(
+    measurement: string,
+    fields: Fields,
+    tags: Tags = {},
+    timestamp: number = nowInSeconds()
+  ): string {
+    let tagString = ''
+    Object.keys(tags)
+      // Sort keys for a little extra perf
+      // https://v2.docs.influxdata.com/v2.0/write-data/best-practices/optimize-writes/#sort-tags-by-key
+      .sort()
```

Comment:
```suggestion
      .sort(((a, b) () => a.localeCompare(b))
```
JavaScripts `sort` array prototype sorts on UTF-16 code units values.  it wont sort lexicographically (alphabetically).